### ISSUE TITLE
not calling AIDelegate

### ISSUE THREAD

**Issue by rakesh-verma-ios** [paytm/Paytm_iOS_All_in_One_SDK#4]
func didFinish(with success: Bool, response: [String : Any]) {
        debugPrint(response)
    }

This delegate function is not called

**Comment by phanindra92** [comment]
I am also facing the issue. Let us know the fix for this.

**Comment by rakesh-verma-ios** [comment]
I have called the transaction status api when pop from Paytm sdk.

**Comment by phanindra92** [comment]
Does it work all the time(With delay and all) ?

**Comment by rakesh-verma-ios** [comment]
Yes when payment is made you will always be popped back, when you will be
popped back then you will hit transaction status api which will tell you
the status of the transaction. We have tested all the cases regarding this.

**Comment by phanindra92** [comment]
This will call delegate method.

1. Create a local variable of AIHandler
example : private var appInvoke : AIHandler = AIHandler()

2. call openPaytm method
example:
 self.appInvoke.openPaytm(merchantId: "put mid here", orderId: orderId, txnToken: txnToken, amount: "100", redirectionUrl : "https://securegw.paytm.in/theia/paytmCallback?ORDER_ID=\(orderId)", delegate: self)

**Comment by rakesh-verma-ios** [comment]
I have used the same but not called

**Comment by divyathakkar123** [comment]
Facing same issue getting message in Paytm that transaction successful and also getting data in Merchant Dashboard. but didn't get any callback in the delegate method. Please help.

Showing following message after completing transaction.
 
**Finish -  https://securegw-stage.paytm.in/theia/paytmCallback?ORDER_ID=B2B45202007280403**

**Comment by rakesh-verma-ios** [comment]
You need to check using the transaction status api for Payment
successful or not.

**Comment by divyathakkar123** [comment]
yes but when can I check that, It should notify in the following method that payment successful.

`
func didFinish(with success: Bool, response: [String : Any]) {
debugPrint(response)
}`

**Comment by rakesh-verma-ios** [comment]
No, whenever payment is successful it will pop to the previous screen where
you can check.

**Comment by divyathakkar123** [comment]
ok thanks.

**Comment by androidroadies** [comment]
@rakesh-verma-ios and @divyathakkar123 Able to resolve issue for get success flag facing same issue.

**Comment by rakesh-verma-ios** [comment]
Yes I was receiving this. I have checked on the previous screen after
payment it will pop. You can add check there and hit getTransaction api to
know the status of the transaction.

**Comment by androidroadies** [comment]
@rakesh-verma-ios Possible to share code snippets specific to how you can check for previous screen as without getting success flag how to identify?

**Comment by androidroadies** [comment]
why can't get flag in didfinish?

**Comment by mohdzakir786** [comment]
Cannot find type 'AIHandler' in scope 

Paytm integration
I am getting this error when I am try to archive project.
